### PR TITLE
[FAB-17601] Fix CI flake due to unexpected pvtdata reconciliation

### PR DESCRIPTION
**Pull request by wenjianqiao**
Signed-off-by: Wenjian Qiao <wenjianq@gmail.com>

#### Type of change
- Test update

#### Description

In private data dissemination test, a test flake may happen if
unexpected pvtdata reconciliation is triggered (timing issue)
and additional peers received the private data.

#### Related issues
https://jira.hyperledger.org/browse/FAB-17601

### CODE REVIEW

File: integration/pvtdata/pvtdata_test.go [hunk 1]
```diff
@@ -779,203 +741,150 @@ var _ bool = Describe("PrivateData", func() {
...
-
-					assertMarbleAPIs()
-					assertDeliverWithPrivateDataACLBehavior()
-				})
-			})
 		})
 
 		Describe("Collection Config Endorsement Policy", func() {
-			When("using legacy lifecycle chaincode", func() {
-				It("ignores the collection config endorsement policy and successfully invokes the chaincode", func() {
```
Comment (denyeart) [inline integration/pvtdata/pvtdata_test.go]: This one actually tests a functional difference between old and new lifecycle.
I think only the tests that are true duplicates across old and new lifecycle (like the other tests above) should be removed.

**Comment by denyeart** [comment]
This PR makes two fundamentally different changes, 1) fix flake due to reconciliation  2) remove duplicate tests across old and new lifecycle.  I'd suggest therefore that it should be split into two separate PRs.  I'd keep this PR for one, and create a different PR for the other.

File: integration/pvtdata/pvtdata_test.go [hunk 2]
```diff
@@ -68,16 +68,17 @@ var _ bool = Describe("PrivateData", func() {
...
-	Describe("Dissemination when pulling is disabled", func() {
+	Describe("Dissemination when pulling and reconciliation are disabled", func() {
 		BeforeEach(func() {
 			By("setting up the network")
 			network = initThreeOrgsSetup(true)
 
-			By("setting the pull retry threshold to 0 on all peers")
-			// set pull retry threshold to 0
+			By("setting the pull retry threshold to 0 and disabling reconciliation on all peers")
+			// disable private data pulling and reconciliation
```
Comment (denyeart) [inline integration/pvtdata/pvtdata_test.go]: Nit - no need for a comment that says the same thing as the By :)

**Review by denyeart — DISMISSED** [review]
Only a nit, so I'll approve.

**Comment by denyeart** [comment]
Hit flake FAB-17611, I'm re-running...

**Comment by wenjianqiao** [comment]
Thanks, Dave. Have addressed your comments.

**Review by denyeart — APPROVED** [review]

**Pull request merged**